pcdsdevices/slits.py

```diff
@@ -58,10 +58,13 @@ class SlitsBase(MvInterface, Device, LightpathMixin):
 
     # Placeholders for each component to override
     # These are expected to be positioners
-    xwidth = None
-    ywidth = None
-    xcenter = None
-    ycenter = None
+    xwidth = Cpt(SignalRO)
+    ywidth = Cpt(SignalRO)
+    xcenter = Cpt(SignalRO)
+    ycenter = Cpt(SignalRO)
+
+    # The gap opens/closes when we move the slits device
+    stage_group = [xwidth, ywidth]
```

Comment:
I see, because this has to refer to a `Component` instance. Ophyd worked around that by accepting either a string name or an instance (if I recall correctly). I think allowing one type as you've done here is probably cleaner.

Semi-related - should there be an `__init_subclass__` for this to ensure that `xwidth` etc are set appropriately? I realize as I'm typing this it's probably overkill, but I'm already near the end of my sentence so 'submit'.

Reply:
Right, I locked it as `Component`-only to encourage lintable code, with the assumption that we wanted to easily catch typos in component names that are harder to catch when you pass in a string.

I think it would make sense to take it a step further, something like making sure they refer to valid components?

Reply:
I like that - the additional check of 'is this a valid component'. 👍